rust/gitxetcore/src/git_integration/git_repo.rs

```diff
@@ -307,7 +308,9 @@ impl GitRepo {
             for r in remotes {
                 s.sync_remote_to_notes(&r)?;
             }
-            s.sync_note_refs_to_local("reposalt", GIT_NOTES_REPO_SALT_REF_SUFFIX)?;
+            // Sync in the notes and mdb stuff as needed.
+            s.sync_notes_to_dbs_for_implicit_init()?;
```

Comment:
I think we can call `sync_notes_to_dbs` instead which does what this function is intended for

Reply:
Turns out we cannot.  The reason is annoying.  That function is async, so we would have to call it through the previous block_on_async function.  However,  libgit2 repo is not send or sync, so it can't be used in the call to that function, which means we have to create a version without the libgit2 repo attribute.  But the merkledb v1 path needs that repo information, so we need to create a path that avoids these constraints.

Reply:
Makes sense